test/common.py

```diff
@@ -935,9 +935,8 @@ def require_jspi(self):
     nodejs = self.get_nodejs()
     if nodejs:
       version = shared.get_node_version(nodejs)
-      # Support for JSPI came earlier than 19, but 19 is what currently works
-      # with emscripten's implementation.
-      if version >= (19, 0, 0):
+      # Support for JSPI came earlier than 22, but the new API changes are not yet in any node
+      if version > (22, 0, 0):
```

Comment:
I think this needs to be `>= (23, 0, 0)` since `(22,0,1) > (22, 0, 0)` will be `True`

Reply:
ah good catch. I guess I could do > 22.99 or something (to reflect that "this is the future" rather than "this is a specific version") but maybe that's too weird.

Reply:
So is node canary version 23?   It looks like we already have  `require_node_canary()` that we could reuse but it also needs to be updated.

Reply:
We should update the node canary we use in circleci to a 23 version too (see https://nodejs.org/download/v8-canary/)

Reply:
OK, so there's a problem with this approach. require_jspi is used for skipping tests when we don't support JSPI but we also want to actually run them under canary. But canary is currently v22 with no indication in the version number that it's really a different node. so that PR fails when it tries to run the JSPI tests under canary.

relatedly there's also require_node_canary which assumes any node > 20 is canary because our "usual" tests use a version less than that.

So probably the "right" behavior for require_jspi for now is to make it just delegate to require_node_canary until there actually exists a non-canary node that has the new API. but that by itself doesn't actually solve the problem, because again there's no indication in the version number.

Reply:
Hence the current version of the patch. It checks the path of nodejs to see if it has the string 'canary' in it. Which is... very hacky. But I don't know of another way to distinguish canary from non-canary, and  it's not much worse than assuming any particular version is canary because that's of course guaranteed to quickly break.